src/composables/accountSelector.ts

```diff
@@ -0,0 +1,240 @@
+        {
+          name: getName(account.address).value || getDefaultAccountLabel(account),
+          address: account.address,
+          isBookmarked: false,
+          protocol: protocolFilter.value ?? PROTOCOLS.aeternity,
+          type: account.type,
+        }
+      ));
+    }
+    return [];
+  });
+  const accountsFilteredByType = computed(
+    () => {
+      switch (accountSelectType.value) {
+        case ACCOUNT_SELECT_TYPE_FILTER.bookmarked:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.addressBook:
+          return addressBookFiltered.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.recent:
+          return latestTransactions.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.owned:
+          return ownAddresses.value;
+        case ACCOUNT_SELECT_TYPE_FILTER.all:
+          return [...(addressBookFiltered.value ?? []), ...(ownAddresses.value ?? [])];
+        default:
+          return [];
+      }
+    },
+  );
+  function filterAccountsBookBySearchPhrase(entries: IAccountSelectorEntry[]) {
+    const searchQueryLower = searchQuery.value.toLowerCase();
+    return entries.filter(({ name, address }) => (
+      [name, address].some((val) => val.toLowerCase().includes(searchQueryLower))
+    ));
+  }
+  function sortAccounts(entries: IAccountSelectorEntry[]) {
+    return entries.sort((a, b) => {
+      if (a === b) return 0;
+      return a ? 1 : -1;
+    });
+  }
+  function removeDuplicates(entries: IAccountSelectorEntry[]) {
+    return entries.filter(
+      (addr1, i, addresses) => addresses.findIndex(
+        (addr2) => addr2.address === addr1.address,
+      ) === i,
+    );
+  }
+  const accountsFiltered = computed(
+    () => pipe([
+      filterAccountsBookBySearchPhrase,
+      sortAccounts,
+      removeDuplicates,
+    ])(accountsFilteredByType.value ?? []),
+  );
+
+  function setAccountSelectType(type: AccountSelectTypeFilter, resetProtocolFilter = false) {
+    accountSelectType.value = type;
+    setShowBookmarked(type === ACCOUNT_SELECT_TYPE_FILTER.bookmarked, resetProtocolFilter);
+  }
+  function clearFilters(resetProtocolFilter = false) {
+    accountSelectType.value = ACCOUNT_SELECT_TYPE_FILTER.addressBook;
+    addressBookClearFilters(resetProtocolFilter);
+  }
+
+  watch(
+    () => [rawAddressBookFiltered.value],
+    () => {
+      addressBookFiltered.value = rawAddressBookFiltered.value.map((account) => (
+        {
+          ...account,
+          isAddressBookEntry: true,
+        }
+      ));
+    },
+    { immediate: true }, // Run immediately on initialization
+  );
+  watch(
+    () => [protocolFilter.value],
+    async () => {
+      if (protocolFilter.value) {
+        setAccountSelectType(ACCOUNT_SELECT_TYPE_FILTER.addressBook);
+      }
+    },
+  );
+  watch(
+    () => [accountsTransactionsLatest.value, activeAccount.value.address],
+    async () => {
+      const filteredTransactions = (accountsTransactionsLatest
```

Comment:
I think `filteredTransactions` const is not need, it should be a part of `.map` below. And we can use `useTransaction` and get all the needed info there.
```
    const {
      innerTx,
      innerTxTag,
      outerTxTag,
      direction,
    } = useTransactionData({
```
Otherwise in case the logic for getting direction would be changed we would need to change it here as well. 

Or we can have a function that will get direction and use it both in `transactionData` composable and here.

Reply:
Updated to use direction from useTransactionData and moved it below